README
NAME
    Plack::Middleware::Status - Plack Middleware for mapping urls to status
    code-driven responses

VERSION
    version 1.101040

SYNOPSIS
        # app.psgi
        use Plack::Builder;
        my $app = sub { 
            # ... 
        };
        builder {
            enable 'Status', path => qr{/not-implemented}, status => 501;
            $app;
        };

AUTHOR
      Patrick Donelan <pat@patspam.com>

COPYRIGHT AND LICENSE
    This software is copyright (c) 2010 by Patrick Donelan.

    This is free software; you can redistribute it and/or modify it under
    the same terms as the Perl 5 programming language system itself.



Changes
Revision history for Task::WebGUI

1.101040  2010-04-14 16:17:40 EST5EDT
    First version



Build.PL

use strict;
use warnings;

use Module::Build 0.3601;

my %module_build_args = (
                       'dist_abstract' => 'Plack Middleware for mapping urls to status code-driven responses',
                       'script_files' => [],
                       'license' => 'perl',
                       'build_requires' => {
                                             'Test::More' => '0',
                                             'Plack::Test' => '0',
                                             'File::Find' => '0',
                                             'File::Temp' => '0',
                                             'Module::Build' => '0.3601',
                                             'Plack::Builder' => '0',
                                             'HTTP::Request::Common' => '0'
                                           },
                       'dist_name' => 'Plack-Middleware-Status',
                       'dist_author' => [
                                          'Patrick Donelan <pat@patspam.com>'
                                        ],
                       'module_name' => 'Plack::Middleware::Status',
                       'requires' => {
                                       'Plack::Middleware' => '0',
                                       'parent' => '0',
                                       'Plack::Util::Accessor' => '0',
                                       'HTTP::Status' => '0',
                                       'Carp' => '0'
                                     },
                       'recommends' => {},
                       'configure_requires' => {
                                                 'ExtUtils::MakeMaker' => '6.31',
                                                 'Module::Build' => '0.3601'
                                               },
                       'dist_version' => '1.101040'
                     );


my $build = Module::Build->new(%module_build_args);

$build->create_build_script;


lib/Plack/Middleware/Status.pm
# 
# This file is part of Plack-Middleware-Status
# 
# This software is copyright (c) 2010 by Patrick Donelan.
# 
# This is free software; you can redistribute it and/or modify it under
# the same terms as the Perl 5 programming language system itself.
# 
package Plack::Middleware::Status;
BEGIN {
  $Plack::Middleware::Status::VERSION = '1.101040';
}

# ABSTRACT: Plack Middleware for mapping urls to status code-driven responses
use strict;
use parent qw/Plack::Middleware/;
use HTTP::Status;
use Plack::Util::Accessor qw( path status );
use Carp;


sub call {
    my $self = shift;
    my $env  = shift;

    my $res = $self->_handle($env);
    return $res if $res;

    return $self->app->($env);
}

sub _handle {
    my ( $self, $env ) = @_;

    my $path_match = $self->path;
    my $status     = $self->status;
    my $path       = $env->{PATH_INFO};
    for ($path) {
        my $matched = 'CODE' eq ref $path_match ? $path_match->($_) : $_ =~ $path_match;
        return unless $matched;
    }

    my $message = HTTP::Status::status_message($status) or do {
        carp "Invalid HTTP status: $status";
        return;
    };

    return [ $status, [ 'Content-Type' => 'text/plain', 'Content-Length' => length($message) ], [$message] ];
}

1;

__END__
=pod

=head1 NAME

Plack::Middleware::Status - Plack Middleware for mapping urls to status code-driven responses

=head1 VERSION

version 1.101040

=head1 SYNOPSIS

    # app.psgi
    use Plack::Builder;
    my $app = sub { 
        # ... 
    };
    builder {
        enable 'Status', path => qr{/not-implemented}, status => 501;
        $app;
    };

=head1 AUTHOR

  Patrick Donelan <pat@patspam.com>

=head1 COPYRIGHT AND LICENSE

This software is copyright (c) 2010 by Patrick Donelan.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.

=cut



Makefile.PL

use strict;
use warnings;



use ExtUtils::MakeMaker 6.31;



my %WriteMakefileArgs = (
                       'test' => {
                                   'TESTS' => 't/*.t'
                                 },
                       'NAME' => 'Plack::Middleware::Status',
                       'DISTNAME' => 'Plack-Middleware-Status',
                       'CONFIGURE_REQUIRES' => {
                                                 'ExtUtils::MakeMaker' => '6.31',
                                                 'Module::Build' => '0.3601'
                                               },
                       'AUTHOR' => 'Patrick Donelan <pat@patspam.com>',
                       'BUILD_REQUIRES' => {
                                             'Test::More' => '0',
                                             'Plack::Test' => '0',
                                             'File::Find' => '0',
                                             'File::Temp' => '0',
                                             'Module::Build' => '0.3601',
                                             'Plack::Builder' => '0',
                                             'HTTP::Request::Common' => '0'
                                           },
                       'ABSTRACT' => 'Plack Middleware for mapping urls to status code-driven responses',
                       'EXE_FILES' => [],
                       'VERSION' => '1.101040',
                       'PREREQ_PM' => {
                                        'Plack::Middleware' => '0',
                                        'parent' => '0',
                                        'Plack::Util::Accessor' => '0',
                                        'HTTP::Status' => '0',
                                        'Carp' => '0'
                                      },
                       'LICENSE' => 'perl'
                     );


delete $WriteMakefileArgs{BUILD_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.56) };

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);



